WCFDataService/Client/System/Data/Services/Client/Binding/BindingGraph.cs

```diff
@@ -1146,6 +1167,46 @@ public bool Equals(Edge other)
                     Object.ReferenceEquals(this.Target, other.Target) &&
                     this.Label == other.Label;
             }
-        }    
+        }
+
+        /// <summary>
+        /// EqualityComparer for Edge's
+        /// </summary>
+        private class EdgeComparer : IEqualityComparer<Edge>
+        {
+            /// <summary>
+            /// Compares two edges for equality.
+            /// </summary>
+            /// <param name="x">Edge x</param>
+            /// <param name="y">Edge y</param>
+            /// <returns>true if they are equal, false otherwise</returns>
+            public bool Equals(Edge x, Edge y)
+            {
+                if (x == null)
+                    return y == null;
+
+                return x.Equals(y);
+            }
+
+            /// <summary>
+            /// Computes the HashCode for an Edge based on Source, Target and Label
+            /// </summary>
+            /// <param name="obj">The Edge to compute the HashCode for</param>
+            /// <returns>The computed HashCode</returns>
+            public int GetHashCode(Edge obj)
+            {
+                if (obj == null)
+                    return 0;
+
+                unchecked
+                {
+                    int hash = 17;
+                    hash = hash * 23 + (obj.Source == null ? 0 : obj.Source.GetHashCode());
```

Comment:
They are just some prime numbers. Basically we want to ensure that (x, y, s) does not return the same hash code as (y, x, s).